src/test/groovy/graphql/schema/somepackage/RecordLikeClass.java

```diff
@@ -0,0 +1,34 @@
+package graphql.schema.somepackage;
+
+import graphql.schema.DataFetchingEnvironment;
+
+/**
+ * This is obviously not an actual record class from Java 14 onwards, but it
+ * smells like one and that's enough really.  Its public, not derived from another
+ * class and has a public method named after a property
+ */
+public class RecordLikeClass {
```

Comment:
Make this `public final class`

Reply:
I have decided we will look up record like methods - not truly final ones like records are.